.buildkite/cache-builder.yml

```diff
@@ -0,0 +1,47 @@
+# This script is run via Buildkite's scheduled jobs feature.
+#
+# It's meant to rebuild various CI caches on a periodic async basis, so as
+# not to waste time on every CI job updating the cache.
+
+# Nodes with values to reuse in the pipeline.
+common_params:
+  # Common plugin settings to use with the `plugins` key.
+  - &common_plugins
+    - automattic/bash-cache#v1.4.0
+    - automattic/git-s3-cache#v1.1.0:
+        bucket: "a8c-repo-mirrors"
+        repo: "wordpress-mobile/wordpress-ios/"
+  # Common environment values to use with the `env` key.
+  - &common_env
+    IMAGE_ID: xcode-12.5.1
+
+steps:
+
+  #################
+  # Build the CocoaPods Base Cache
+  # 
+  # This prevents the base cache from infinite growth caused by storing every
+  # version of every pod we've ever used.
+  #################
+  - label: ":cocoapods: Rebuild CocoaPods cache"
+    command: |
+      echo "--- :rubygems: Setting up Gems"
+      install_gems
+
+      echo "--- :cocoapods: Rebuilding Pod Cache"
+      cache_cocoapods
```

Comment:
Had to look at the implementation and especially [what directories were saved](https://github.com/Automattic/bash-cache-buildkite-plugin/blob/trunk/bin/cache_cocoapods#L7-L8) to realize this doesn't cache the `Pods/` directory from the repo (†) but instead cache the `~/.cocoapods` and `~/Library/Caches/CocoaPods/` cache, which makes way more sense as a scheduled pipeline run overnight.

I wonder if we could better document this to avoid the ambiguity.

(†) And I would have had concerns if it were that case because it'd then only cache the `Pods/` dir from `develop` then, and cleaning the cached `Pods/` from PRs and branches that modify the `Podfile.lock`

Reply:
Hmm yeah – ideally I'd like this to be self-documenting, so WDYT about changing the `cache_cocoapods` action to something like `cache_cocoapods_specs_repos` or similar?

Reply:
Yep, that was gonna be my next suggestion indeed.

I think the confusing thing is that "Pod" / "Pods" / "Pod Cache" is currently used interchangeably with "cocoapods", for example the section is called "Pod Cache" with is probably what made me think of `Pods/` instead, and also `cache_cocoapods` and `install_cocoapods` both use "cocoapods" in their name while one cache the pods repositories and the `~/Library/Cache/Pods` which are system-wide things, while the other install the actual pods in the `Pods/` directory of the repo.

Changing that to `cache_cocoapods_specs_repos` would solve that indeed 👍  – though… this command also caches `~/Library/Caches/CocoaPods/` not just the spec repo at `~/.cocoapods`… damn, naming is hard.

Reply:
Addressed in 1b12300b22ffca7d63e475206dab545e8907399c